### ISSUE TITLE
Possibly fixed overfitting?

### ISSUE THREAD

**Issue by KoleGuenther** [KoleGuenther/ibl#2]
Current adam solver has this learning curve graph:
![image](https://github.com/user-attachments/assets/90acefa9-0db7-42a0-a2b8-eda27f92aeec)

Here is the model performance otherwise:
![image](https://github.com/user-attachments/assets/9c9ff6b3-4e0f-4b69-93de-ae0c7ae2b0f0)
Accuracy : 91.95%
Precision : 90.27%
Recall : 99.03%
F1 Score : 94.44%

StratifiedKFold:
Cross-Validation Accuracy Scores (for each fold): [0.92462312 0.92964824 0.93434343 0.93434343 0.88888889]
Mean Cross-Validation Accuracy: 0.9224
Standard Deviation of Cross-Validation Accuracy: 0.0171

Here is the settings used:
solver='adam',
hidden_layer_sizes=(15, 8,),
max_iter=4000,
alpha=0.5,
activation='relu',
tol=1e-5,
random_state=4,
learning_rate_init=0.005
Threshold: 0.45
